backend/src/main/java/com/timlenny/backend/service/TopicService.java

```diff
@@ -40,12 +40,36 @@ public Topic addTopic(TopicDTO topicToAddDTO) {
         }
 
         Optional<Topic> parentTopic = topicRepository.findByTopicName(topicToAddDTO.getParentName());
-        if (parentTopic.isPresent()) {
-            Topic topicToAdd = topicConversionService.convertNewDTOtoTopic(topicToAddDTO, parentTopic.get());
-            topicRepository.save(topicToAdd);
-            return topicToAdd;
-        } else {
-            throw new ResponseStatusException(HttpStatus.NO_CONTENT, "parentTopic not found!");
+
+        if (parentTopic.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "parentTopic not found!");
+        }
+
+        Topic topicToAdd = topicConversionService.convertNewDTOtoTopic(topicToAddDTO, parentTopic.get());
+        topicRepository.save(topicToAdd);
+        return topicToAdd;
```

Comment:
`.save` already returns the saved object :)
```suggestion
        return topicRepository.save(topicToAdd);
```